resource/interface_factories.cpp.em

```diff
@@ -332,3 +337,197 @@ void ServiceFactory<
 @[  end for]@
 @[end for]@
 }  // namespace ros1_bridge
+
+
+// ROS1 serialization functions
+namespace ros1_bridge
+{
+
+// This version is for write or length
+template<typename STREAM_T, typename VEC_T>
+static void streamVectorSize(STREAM_T& stream, const VEC_T& vec)
+{
+  // Output size of vector to stream
+  uint32_t data_len = vec.size();
+  stream.next(data_len);
+}
+
+// This version is for read
+template<typename STREAM_T, typename VEC_T>
+static void streamVectorSize(STREAM_T& stream, VEC_T& vec)
+{
+  // Resize vector to match size in stream
+  uint32_t data_len = 0;
+  stream.next(data_len);
+  vec.resize(data_len);
+}
+
+// This version is for write
+template<typename VEC_PRIMITIVE_T>
+static void streamPrimitiveVector(ros::serialization::OStream & stream, const VEC_PRIMITIVE_T& vec)
+{
+  const uint32_t data_len = vec.size() * sizeof(typename VEC_PRIMITIVE_T::value_type);
+  // copy data from std::vector/std::array into stream
+  memcpy(stream.advance(data_len), &vec.front(), data_len);
+}
+
+// This version is for length
+template<typename VEC_PRIMITIVE_T>
+static void streamPrimitiveVector(ros::serialization::LStream & stream, const VEC_PRIMITIVE_T& vec)
+{
+  const uint32_t data_len = vec.size() * sizeof(typename VEC_PRIMITIVE_T::value_type);
+  stream.advance(data_len);
+}
+
+// This version is for read
+template<typename VEC_PRIMITIVE_T>
+static void streamPrimitiveVector(ros::serialization::IStream & stream, VEC_PRIMITIVE_T& vec)
+{
+  const uint32_t data_len = vec.size() * sizeof(typename VEC_PRIMITIVE_T::value_type);
+  // copy data from stream into std::vector/std::array
+  memcpy(&vec.front(), stream.advance(data_len), data_len);
+}
+
+@[for m in mapped_msgs]@
+
+@[  if m.ros2_msg.package_name=="std_msgs" and m.ros2_msg.message_name=="Header"]
+// msg_2_to_1_stream for std_msgs::(msg)::Header should be specialized because of lack of seq
```

Comment:
Is this included in this PR?

Reply:
Yes, the template specialization is part of builtin_interfaces_factories.cpp   
This is similar to template specialization for Duration and Time